sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/StorageImplUtils.java

```diff
@@ -485,4 +490,67 @@ public static String getEncryptionDataKey(Map<String, String> metadata) {
         }
         return null;
     }
+
+    public static <T, U> ResponseBase<T, U> sendRequest(Callable<ResponseBase<T, U>> operation, Duration timeout) {
+        try {
+            Future<ResponseBase<T, U>> future = THREAD_POOL.submit(operation);
+            if (timeout == null) {
+                return getResultWithTimeout(future, 0, RuntimeException.class);
+            }
+            return getResultWithTimeout(future, timeout.toMillis(), RuntimeException.class);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            throw LOGGER.logExceptionAsError(new RuntimeException(e));
+        }
+    }
+
+    public static <T, U> ResponseBase<T, U> sendRequest(Callable<ResponseBase<T, U>> operation, Duration timeout, Class<? extends RuntimeException> exceptionType) {
+        try {
+            Future<ResponseBase<T, U>> future = THREAD_POOL.submit(operation);
+            if (timeout == null) {
+                return getResultWithTimeout(future, 0, exceptionType);
+            }
+            return getResultWithTimeout(future, timeout.toMillis(), exceptionType);
+        } catch (InterruptedException | ExecutionException | TimeoutException e) {
+            Throwable cause = e.getCause();
+            if (exceptionType.isInstance(cause)) {
+                // Safe to cast since we checked with isInstance
+                throw exceptionType.cast(cause);
+            } else if (cause instanceof RuntimeException) {
+                // Throw as is if it's already a RuntimeException
+                throw (RuntimeException) cause;
+            } else if (cause instanceof Error) {
+                // Propagate if it's an Error
+                throw (Error) cause;
+            } else {
+                // Wrap in RuntimeException if it's neither Error nor RuntimeException
+                throw LOGGER.logExceptionAsError(new RuntimeException("Exception during async operation", cause));
```

Comment:
nit: I'd drop the message in the `RuntimeException` or slightly change it.

`Exception during synchronous operation with timeout`, as this shouldn't be called for asynchronous operations, only sync with timeout